homeassistant/components/hyperion/config_flow.py

```diff
@@ -0,0 +1,379 @@
+"""Hyperion config flow."""
+
+import asyncio
+import logging
+from typing import Any, Dict, Optional
+from urllib.parse import urlparse
+
+from hyperion import client, const
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.ssdp import ATTR_SSDP_LOCATION, ATTR_UPNP_SERIAL
+from homeassistant.const import CONF_BASE, CONF_HOST, CONF_ID, CONF_PORT, CONF_TOKEN
+from homeassistant.helpers.typing import ConfigType
+
+# pylint: disable=unused-import
+from .const import CONF_AUTH_ID, CONF_CREATE_TOKEN, DEFAULT_ORIGIN, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.DEBUG)
+
+#  +------------------+    +------------------+    +--------------------+
```

Comment:
Will this comment really be kept updated as the integration evolves?

Reply:
I appreciate this question. 

I think the diagram makes it easier for people not familiar with the flow to understand what's happening. The Hyperion component has barely changed in the past years (except large scale codebase changes) until I became the only code owner a few weeks back.

In short: I think it's better with the diagram. The codebase rarely ever changes. I think it's worth trying to keep it up to date!